config/v1/types_infrastructure.go

```diff
@@ -567,13 +578,35 @@ type GCPPlatformStatus struct {
 	Region string `json:"region"`
 }
 
+// BareMetalPlatformLoadBalancer defines the load balancer used by the cluster on BareMetal platform.
+// +union
+type BareMetalPlatformLoadBalancer struct {
+	// type defines the type of load balancer used by the cluster on BareMetal platform
+	// which can be a user-managed or openshift-managed load balancer
+	// that is to be used for the OpenShift API and Ingress endpoints.
+	// When set to OpenShiftManagedDefault the static pods in charge of API and Ingress traffic load-balancing
+	// defined in the machine config operator will be deployed.
+	// When set to UserManaged these static pods will not be deployed and it is expected that
+	// the load balancer is configured out of band by the deployer.
+	// When omitted, this means no opinion and the platform is left to choose a reasonable default.
+	// The default value is OpenShiftManagedDefault.
+	// +default="OpenShiftManagedDefault"
+	// +kubebuilder:default:="OpenShiftManagedDefault"
+	// +kubebuilder:validation:Enum:="OpenShiftManagedDefault";"UserManaged"
+	// +kubebuilder:validation:XValidation:rule="oldSelf == '' || self == oldSelf",message="type is immutable once set"
+	// +optional
+	// +unionDiscriminator
+	Type PlatformLoadBalancerType `json:"type,omitempty"`
+}
+
 // BareMetalPlatformSpec holds the desired state of the BareMetal infrastructure provider.
 // This only includes fields that can be modified in the cluster.
 type BareMetalPlatformSpec struct{}
 
 // BareMetalPlatformStatus holds the current status of the BareMetal infrastructure provider.
 // For more information about the network architecture used with the BareMetal platform type, see:
 // https://github.com/openshift/installer/blob/master/docs/design/baremetal/networking-infrastructure.md
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.loadBalancer) && !has(self.loadBalancer) || has(oldSelf.loadBalancer) && has(self.loadBalancer)",message="loadBalancer may only be configured during installation"
```

Comment:
So reviewing the test cases, I think this may not actually be needed, it seems the defaulting is happening before the validation, so if you remove the entire loadBalancer struct, it then causes the default to kick in, which then is passing validation, that's a little irritating as it means if the value is the default, and someone tries to unset `loadBalancer`, we can't error and tell them no, you can't do that, the only reason it works is because the default is the same

Can you try a test case with the default value, and unsetting that, what happens? If it passes, and doesn't error, then I can't see a way for this validation to actually be hit, so we may as well remove it